src/subscription/index.ts

```diff
@@ -0,0 +1,43 @@
+import type { Extension } from '../types'
+import { useIpc } from '../ipc/index'
+
+interface ProSubscriptionUpsellOptions {
+  firstpromoter?: {
+    slug?: string
+  }
+}
+
+interface SubscriptionComposable {
+  showProSubscriptionUpsell: (options: ProSubscriptionUpsellOptions) => void
+}
+
+/**
+ * The Subscription module provides methods to the Subscription.
+ *
+ * @param extension - The extension instance
+ *
+ * @example
+ * import { initializeExtension } from '@own3d/sdk/extension'
+ * import { useSubscription } from '@own3d/sdk/subscription'
+ *
+ * const extension = initializeExtension()
+ *
+ * const { showProSubscriptionUpsell } = useSubscription(extension)
+ *
+ * showProSubscriptionUpsell({
+ *  firstpromoter: {
+ *   slug: 'my-slug',
+*   }
+ * })
+ */
+export function useSubscription(extension: Extension): SubscriptionComposable {
+  const { send } = useIpc(extension)
+
+  function showProSubscriptionUpsell(options: any) : void {
+    send('subscription.own3d.pro/show-pro-sub-upsell', options)
```

Comment:
Nur für die Records:

Die Idee der "dns names" ist in erster Linie nur dazu da, um Namensräume zu schaffen. Ich würde es aber vermeiden, das wir zu viele "domains" erstellen. Aktuell hab ich mich auf `scene-builder.own3d.pro`, `obsproject.com/` sowie `desktop.own3d.pro/` beschränkt. So wissen wir grob, wohin/woher die Events kommen.

Ich hätte jetzt z.B. nur `own3d.pro/` genommen, so wissen wir, dass alle Events darin für das OWN3D Pro Frontend bestimmt sind. Und wenn weitere 3rd Party Events hinzukommen, können sie einfach Ihre Domain `[app.]example.com/` als "Channel" hinzufügen.